src/validators/dataclass.rs

```diff
@@ -449,6 +432,46 @@ impl Validator for DataclassValidator {
     fn get_name(&self) -> &str {
         &self.name
     }
+
+    fn validate_assignment<'s, 'data: 's>(
+        &'s self,
+        py: Python<'data>,
+        obj: &'data PyAny,
+        field_name: &'data str,
+        field_value: &'data PyAny,
+        extra: &Extra,
+        slots: &'data [CombinedValidator],
+        recursion_guard: &'s mut RecursionGuard,
+    ) -> ValResult<'data, PyObject> {
+        let dict: &PyDict = match obj.get_attr(intern!(py, "__dict__")) {
+            Some(v) => v.downcast()?,
+            None => {
+                return Err(ValError::InternalErr(PyTypeError::new_err(format!(
+                    "{} is not a model instance",
+                    obj.repr()?
+                ))))
+            }
+        };
+
+        let new_dict = dict.copy()?;
+        new_dict.set_item(field_name, field_value)?;
+
+        let new_input: &PyAny = new_dict;
```

Comment:
you shouldn't need this line, I think you can use `new_dict as &PyAny` or `new_dict.as_ref()`.